java/ql/src/experimental/Security/CWE/CWE-552/UnsafeResourceGet.java

```diff
@@ -0,0 +1,16 @@
+// BAD: no URI validation
+URL url = servletContext.getResource(requestUrl);
+url = getClass().getResource(requestUrl);
+InputStream in = url.openStream();
+
+InputStream in = request.getServletContext().getResourceAsStream(requestPath);
+in = getClass().getClassLoader().getResourceAsStream(requestPath);
+
+// GOOD: check for a trusted prefix, ensuring path traversal is not used to erase that prefix:
+// (alternatively use `Path.normalize` instead of checking for `..`)
+if (!requestPath.contains("..") && requestPath.startsWith("/trusted")) {
+	InputStream in = request.getServletContext().getResourceAsStream(requestPath);
+}
+
+Path path = Paths.get(requestUrl).normalize().toRealPath();
```

Comment:
Isn't this lacking the `startsWith` check (or similar) to be considered secure?

Reply:
Yes, it lacks the `startsWith` check and I've added it in the sample usage.